tools/generator/AnDOGenerator.py

```diff
@@ -10,6 +15,117 @@
            ]
 
 
+class AnDOSesID:
+
+    def __init__(self, sesID=None, date=None, sesNumber=None, customSesField=None):
+        """
+        Representation of an AnDO session ID, as defined in `ando/rules/session_rules.json`
+
+        Parameters
+        ----------
+        sesID: (str)
+            complete session identifier, e.g. `20000101_001_test`
+        date: (str or datetime)
+            date subinfo of session ID, e.g. `20000101`
+        sesNumber: (str)
+            session number subinfo of session ID, e.g. `001`
+        customSesField: (str)
+            custom session field subinfo of session ID
+        """
+
+        # Inform user if insufficient or duplicate information is provided
+        if sesID is not None:
+            if (date is None) or (sesNumber is None) or (customSesField is None):
+                warnings.warn(f'Using sesID ({sesID}) in favour alternative parameters date, '
+                              f'sesNumber and customSesField')
+        elif (date is None) or (sesNumber is None):
+            raise ValueError('Incomplete information for generating a session ID.')
+
+
+        if sesID is None:
+            self.sesNumber = sesNumber
+
+            # Define default value for custom session field
+            if customSesField is None:
+                self.customSesField = ''
+            else:
+                self.customSesField = customSesField
+
+            if isinstance(date, datetime.datetime):
+                self.date = date.strftime('%Y%m%d')
+            else:
+                self.date = date
+
+        # if only sesID is provided, we use the session regex to extract infos
+        else:
+            rules_dir = os.path.join(os.path.dirname(ando.__file__), 'rules')
```

Comment:
Yes I agree, maybe it would be even good to have a wrapper that provides the rules directly, so we are not loading them in x different ways across our code.

Reply:
However, this should go into another PR, since this one is already quite complex. I opened #54 as a reminder